src/scheduler.cc

```diff
@@ -293,29 +307,25 @@ Status SchedulerService::KillWorkers(ServerContext* context, const KillWorkersRe
 
 Status SchedulerService::ExportFunction(ServerContext* context, const ExportFunctionRequest* request, ExportFunctionReply* reply) {
   auto workers = workers_.get();
+  auto exported_functions = exported_functions_.get();
+  // TODO(rkn): Does this do a deep copy?
```

Comment:
we can be more sure about that if we create the unique_ptr from a new Function() via

std::unique_ptr<Function> function(new Function());
function->CopyFrom(request->function())

and then pass it in; it is a little more verbose though!